### PR TITLE
Migrate to using 3P units library for everything

### PR DESCRIPTION
**Pull request by kammce**
Move to using user defined literals for all uses of units like
frequency, distance, time, etc.

### CODE REVIEW

File: library/L1_Peripheral/lpc17xx/system_controller.hpp [hunk 1]
```diff
@@ -386,29 +380,29 @@ class SystemController final : public sjsu::SystemController
...
+  ///                        exceed the maximum allowed CPU clock.
+  Pll0Settings_t CalculatePll0(units::frequency::hertz_t input_frequency,
+                               units::frequency::hertz_t desired_speed) const
   {
     // minimum/maximum input and output frequencies of PLL0 in kHz
-    constexpr uint32_t kMinimumPll0InputFrequency = 32;
-    constexpr uint32_t kMaximumPll0InputFrequency = 50'000;
-    constexpr uint32_t kMinimumPll0OuputFrequency = 275'000;
-    constexpr uint32_t kMaximumPll0OuputFrequency = 550'000;
+    constexpr units::frequency::hertz_t kMinimumPll0InputFrequency = 32_Hz;
```
Comment (HuangDave) [inline library/L1_Peripheral/lpc17xx/system_controller.hpp]: The kMinimumPll0InputFrequency should be in kHz

Comment (HuangDave) [inline library/L1_Peripheral/lpc17xx/system_controller.hpp]: kMinimumPll0InputFrequency needs to be be in kHz.

File: library/L1_Peripheral/lpc17xx/system_controller.hpp [hunk 2]
```diff
@@ -386,29 +380,29 @@ class SystemController final : public sjsu::SystemController
...
+    constexpr units::frequency::hertz_t kMaximumPll0InputFrequency = 50'000_Hz;
+    constexpr units::frequency::hertz_t kMinimumPll0OuputFrequency = 275'000_Hz;
```
[the remark this answers is not in the document]
Comment (kammce) [reply library/L1_Peripheral/lpc17xx/system_controller.hpp]: Done.

File: library/L1_Peripheral/lpc17xx/system_controller.hpp [hunk 3]
```diff
@@ -386,29 +380,29 @@ class SystemController final : public sjsu::SystemController
...
+  Pll0Settings_t CalculatePll0(units::frequency::hertz_t input_frequency,
+                               units::frequency::hertz_t desired_speed) const
   {
     // minimum/maximum input and output frequencies of PLL0 in kHz
-    constexpr uint32_t kMinimumPll0InputFrequency = 32;
-    constexpr uint32_t kMaximumPll0InputFrequency = 50'000;
-    constexpr uint32_t kMinimumPll0OuputFrequency = 275'000;
-    constexpr uint32_t kMaximumPll0OuputFrequency = 550'000;
+    constexpr units::frequency::hertz_t kMinimumPll0InputFrequency = 32_Hz;
+    constexpr units::frequency::hertz_t kMaximumPll0InputFrequency = 50'000_Hz;
```
[the remark this answers is not in the document]
Comment (kammce) [reply library/L1_Peripheral/lpc17xx/system_controller.hpp]: Done. Thanks for the heads up about this, makes more sense now.

File: library/L1_Peripheral/lpc17xx/system_controller.hpp [hunk 4]
```diff
@@ -421,19 +415,25 @@ class SystemController final : public sjsu::SystemController
...
       {
-        // current controlled oscilator frequency, fcco, output of PLL0
-        const uint32_t kFcco = (2 * (m + 1) * input_frequency_in_khz) / (n + 1);
-        if (kFcco > kMinimumPll0OuputFrequency &&
+        // Current calculated controlled oscillator frequency, fcco, output of
+        // PLL0 in kilohertz
+        // Dividing by 1000 to scale down kFcco, as the multiplier for is
+        // internally scaled by 1000.
+        const units::frequency::hertz_t kFcco =
+            (2 * (m + 1) * input_frequency / 1000) / (n + 1);
```
[the remark this answers is not in the document]
Comment (kammce) [reply library/L1_Peripheral/lpc17xx/system_controller.hpp]: Done.

**Pull request merged**